src/azul/chalice.py

```diff
@@ -467,6 +494,32 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
+    def threshold(self, *, errors: int, throttles: int):
+        def wrapper(f):
+            f.errors_threshold = errors
+            f.throttles_threshold = throttles
+            return f
+
+        return wrapper
+
+    def metric_thresholds(self) -> list[MetricThreshold]:
+        default_threshold = 0
+        thresholds = []
+        lambda_name, _ = config.unqualified_resource_name(self.app_name)
+        for metric in LambdaMetric:
+            # The api_handler lambda functions (indexer & service) aren't included
+            # in the app_module's handler_map, so we account for those first.
+            thresholds.append(MetricThreshold(lambda_name=lambda_name,
+                                              metric=metric,
+                                              value=default_threshold))
+            for handler_name, handler in self.handler_map.items():
+                if isinstance(handler, chalice.app.EventSourceHandler):
+                    threshold = getattr(handler, f'{metric.name}_threshold', default_threshold)
+                    thresholds.append(MetricThreshold(lambda_name=f'{lambda_name}_{handler_name}',
```

Comment:
The `_` separator is a TF-specific detail. Calling the attribute `lambda_name` is really confusing because it conflicts with the meaning we have for it everywhere else. Even in your new code, the different meanings collide. I think you should track lambda and handler name separately in MetricThreshold.